_posts/2016-03-01-Docker-meets-Continuous-Deployment.md

```diff
@@ -0,0 +1,179 @@
+We will define later elements of this directory. 
+
+We need two artifacts. The first one is `Dockerfile`. We already prepared this file in our source code and it will be stored in `docker` directory.
+The next file is ***tar*** file which will be generated by our **Gradle** build. It contains script which allows to execute our code and all required libraries. 
+
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/build_config.png" alt="Build Configuration" title="Build Configuration" class="img img-responsive style-screengrab">
+
+Now we are ready to instruct our build configuration how to download our source code. We use github as our code repository.
+We just have to choose type as *Git* and provide url to our application.
+
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/build_vcs.png" alt="Version Control Configuration" title="Version Control Configuration" class="img img-responsive style-screengrab">
+
+We have the source code, so we can describe what to do with this code. As I mention before we will **test** our application and generate an **artifact** with executable content.   
+
+**TeamCity** have predefined runners for different build tools. In this case we are going to use **Gradle** runner to **test** and **build**. 
+
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/build_step.png" alt="Gradle Step" title="Gradle Step" class="img img-responsive style-screengrab">
+
+Now we can run our build. As result of we should see our **artifacts** in *"Artifact"* tab. 
+ 
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/build_result.png" alt="Build Result" title="Build Result" class="img img-responsive style-screengrab">
+
+## Release Image
+
+Our code is no longer needed. We have all we need to build the docker image. Now we have create our image and **release** with the right version. To simplify our example we are going to use current build number to define an image version.
+Next we will generate file with this version. This file give us possibility to pass information about version to the next steps.  
+
+First, let's take a closer look at `Dockerfile`. We copy to the image all libraries and scripts from `simple_application.tar`. This ***tar*** file already contains all required content, so all we have to do is to use `ADD`. It will automatically *untar* all files in the image. Next we expose port of our HTTP API and we define how we launch our application by adding `ENTRYPOINT` command.  
+
+For example our `Dockerfile` can look like as follows: 
+
+```
+FROM java:8
+
+ADD simple_application.tar .
+
+EXPOSE 4567
+
+ENTRYPOINT ["/simple_application/bin/simple_application"]
+```
+
+Definition of the build configuration is very simple. In the general settings we just define new artifact: `image.version`. We will generate this file in a build step and it will contain an docker image version.
```

Comment:
Defining the build configuration is very simple. In the general settings we just define a new artefact: ...
We will generate this file in a build step and it will contain a docker image version.